onnxruntime/core/mlas/lib/intrinsics/avx2/qladd_avx2.cpp

```diff
@@ -75,6 +75,30 @@ MlasPackS16_256<int8_t>(
     return _mm256_packs_epi16(a, b);
 }
 
+MLAS_FORCEINLINE
+static
+__m256i
+MlasLoad32Bytes(const uint8_t* buffer, int64_t N)
+{
+    if (N >= 32) {
+        return _mm256_lddqu_si256((const __m256i*)buffer);
+    } else {
+        uint8_t dup[32];
```

Comment:
>32 [](start = 20, length = 2)

This will have garbage at the end?

Reply:
yes, it is on stack.

Reply:
So you are copying more data than present in the src buffer. Is the destination buffer large enough to recieve extra data?

---
In reply to: [451845799](https://github.com/microsoft/onnxruntime/pull/4455#discussion_r451845799) [](ancestors = 451845799)